src/libcollections/slice.rs

```diff
@@ -1041,8 +1041,11 @@ impl<T> [T] {
 
     /// This is equivalent to `self.sort_by(|a, b| a.cmp(b))`.
     ///
-    /// This sort is stable and `O(n log n)` worst-case, but allocates
-    /// temporary storage half the size of `self`.
+    /// This sort is stable and `O(n log n)` worst-case. 
+    ///
+    /// # Current Implementation
+    /// 
+    /// The current implementation allocates temporary storage half the size of `self`.
```

Comment:
Now that we have a section about the current implementation, this would be a great chance to expand a little bit. For inspiration, see [Java's docs](https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#sort(java.util.List)) starting from "Implemenation note".

I would like to see several things pointed out:

1. That this sort is an adaptation of Tim Peter's timsort with this [link](http://svn.python.org/projects/python/trunk/Objects/listsort.txt).
2. That this is an iterative sort.
3. That this is an adaptive sort. Again, Java has a nice explanation: "It is well-suited to merging two or more sorted arrays: simply concatenate the arrays and sort the resulting array."